File: src/main/java/kr/devport/api/DevportApiApplication.java
package kr.devport.api;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.scheduling.annotation.EnableAsync;
import org.springframework.scheduling.annotation.EnableScheduling;

@SpringBootApplication
@EnableAsync
@EnableScheduling
public class DevportApiApplication {

    public static void main(String[] args) {
        SpringApplication.run(DevportApiApplication.class, args);
    }
}

File: application-api/src/main/kotlin/kr/devport/api/DevportApiApplication.kt
package kr.devport.api

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.persistence.autoconfigure.EntityScan
import org.springframework.boot.runApplication
import org.springframework.data.jpa.repository.config.EnableJpaRepositories
import org.springframework.scheduling.annotation.EnableAsync
import org.springframework.scheduling.annotation.EnableScheduling

/**
 * Composition root. Component, entity, and repository scanning is explicit so the Kotlin
 * domain modules under `kr.devport.api` and the gitrepo modules under `kr.devport.gitrepo`
 * are composed into one bootable application.
 */
@SpringBootApplication
@EnableAsync
@EnableScheduling
@EntityScan(basePackages = ["kr.devport.api", "kr.devport.gitrepo"])
@EnableJpaRepositories(basePackages = ["kr.devport.api", "kr.devport.gitrepo"])
class DevportApiApplication

fun main(args: Array<String>) {
    runApplication<DevportApiApplication>(*args)
}
